spec/crimp_spec.rb

```diff
@@ -1,99 +1,299 @@
-        original_hash = { d: 'd', a: { c: 'c', b: 'b' } }
-        expected_hash = { d: 'd', a: { c: 'c', b: 'b' } }
-
-        subject.signature(original_hash)
-
-        expect(original_hash).to eq(expected_hash)
-      end
-    end
-
-    context 'Given an hash with numbers' do
-      it 'returns MD5 hash of stringified hash' do
-        expect(subject.signature(hash_with_numbers)).to eq 'b1fec09904b6ff36c92e3bd48234def7'
-      end
-    end
-
-    context 'given an Array' do
-      it 'returns MD5 hash of stringified Array' do
-        expect(subject.signature(array)).to eq('4dc4e1161c9315db0bc43c0201b3ec05')
-      end
-
-      it 'does not modify original array' do
-        original_array = [5, 4, 2, 6, [5, 7, 2]]
-        expected_array = [5, 4, 2, 6, [5, 7, 2]]
-
-        subject.signature(original_array)
-
-        expect(original_array).to eq(expected_array)
-      end
-    end
-
-    context 'Given an integer' do
-      it 'returns MD5 hash of an Integer' do
-        expect(subject.signature(123)).to eq '519d3381631851be66711f6d7dfbb4f8'
-      end
-    end
-
-    context 'Given an Bignum' do
-      it 'returns MD5 hash of a Bignum' do
-        expect(subject.signature(9999999999999999999)).to eq 'f00e75abca720e18fd4213e2a6de96c6'
-      end
-    end
-
-    context 'Given an float' do
-      it 'returns MD5 hash of a Float' do
-        expect(subject.signature(3.14)).to eq 'b07d506e3701fddd083ae9095df43218'
-      end
-    end
-  end
-
-  describe '.stringify' do
-    context 'given a Hash' do
-      it 'returns equal strings for differently ordered hashes' do
-        expect(subject.stringify(hash)).to eq(subject.stringify(hash_unordered))
-      end
-
-      it 'does not modify original hash' do
-        original_hash = { d: 'd', a: { c: 'c', b: 'b' } }
-        expected_hash = { d: 'd', a: { c: 'c', b: 'b' } }
-
-        subject.signature(original_hash)
-
-        expect(original_hash).to eq(expected_hash)
-      end
-    end
-
-    context 'given an Array' do
-      specify { expect(subject.stringify(array)).to be_a String }
-
-      it 'returns equal strings for differently ordered arrays' do
-        expect(subject.stringify(array)).to eq(subject.stringify(array_unordered))
-      end
-
-      it 'does not modify original array' do
-        original_array = [5, 4, 2, 6, [5, 7, 2]]
-        expected_array = [5, 4, 2, 6, [5, 7, 2]]
-
-        subject.signature(original_array)
-
-        expect(original_array).to eq(expected_array)
-      end
-    end
+describe '.signature' do
+  it 'will return an md5 hash' do
+    expect(Crimp.signature('a')).to eq 'd132c0567a5964930f9ee5f14e779e32'
+  end
+end
+
+describe '.notation' do
+  it 'returns a string representation of the passed data' do
+    expect(Crimp.notation([123, 'abc'])).to eq('123NabcSA')
+  end
+end
+
+describe '.reduce' do
+  it 'returns an array of tuples representing the value and the type' do
+    expect(Crimp.reduce([123, 'abc'])).to eq([[[123, 'N'], ['abc', 'S']], 'A'])
```

Comment:
Not sure this is reducing the input as it's adding notation values and now seems larger.

Reply:
categorise ?

Reply:
thinking the same... it's not reducing. better name?

Reply:
categorising, labelling, classifying, tagging or indexing.

Reply:
`Crimp.to_tuple`,'`Crimp.build`,`Crimp.to_sequence`,`Crimp.concat`,`Crimp.stack`

Reply:
I like categorise, tuple, or sequence. Or perhaps `Crimp.scaffold` or `Crimp.structure`

Reply:
annotate? label?

Reply:
annotate